pad_server/src/actuator.c

```diff
@@ -38,18 +49,42 @@ void actuator_init(actuator_t *act, uint8_t id, actuate_f on, actuate_f off, voi
  * @param act The actuator to turn on.
  * @return 0 for success, an error code on failure.
  */
-int actuator_on(actuator_t *act) { return act->on(act); }
+int actuator_on(actuator_t *act) { return 0; }
 
 /*
  * Turn the actuator off.
  * @param act The actuator to turn off.
- * @return 0 for success, an error code on failure.
+ * @return 0 for an error code on failure.
  */
-int actuator_off(actuator_t *act) { return act->off(act); }
+int actuator_off(actuator_t *act) { return 0; }
 
 /*
  * Get the string name of the actuator.
  * @param act The actuator to get the string name of.
  * @return The string name of the actuator.
  */
-const char *actuator_name(actuator_t *act) { return ACTUATOR_STR[act->id]; }
+const char *actuator_get_name(actuator_t *act) { return ACTUATOR_STR[act->id]; }
+
+/*
+ * Set the value of an actuator
+ * @param id The actuator id
+ * @param req_state The new actuator state
+ * @return ACT_OK for success, ACT_DNE for invalid id, ACT_INV for invalid req_state, -1 for errors eith errno being
+ * set
+ */
+int actuator_set(padstate_t *state, uint8_t id, bool new_state) {
```

Comment:
This function doesn't need the state, just the pointer to an actuator. It should operate only on the actuator.

If you want to operate on the state you should create a function in the padstate to call `actuator_set` after locking the wrlock.